v2/pubsub-cdc-to-bigquery/src/test/java/com/google/cloud/teleport/v2/templates/PubSubCdcToBigQueryIT.java

```diff
@@ -0,0 +1,236 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.it.common.PipelineLauncher;
+import org.apache.beam.it.common.PipelineLauncher.LaunchConfig;
+import org.apache.beam.it.common.PipelineOperator;
+import org.apache.beam.it.common.utils.ResourceManagerUtils;
+import org.apache.beam.it.gcp.TemplateTestBase;
+import org.apache.beam.it.gcp.bigquery.BigQueryResourceManager;
+import org.apache.beam.it.gcp.bigquery.conditions.BigQueryRowsCheck;
+import org.apache.beam.it.gcp.pubsub.PubsubResourceManager;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.json.JSONObject;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@Category(TemplateIntegrationTest.class)
+@TemplateIntegrationTest(PubSubCdcToBigQuery.class)
+@RunWith(JUnit4.class)
+public class PubSubCdcToBigQueryIT extends TemplateTestBase {
+
+  private static final int MESSAGES_COUNT = 10;
+  private static final int BAD_MESSAGES_COUNT = 3;
+
+  private static final Schema BIG_QUERY_DLQ_SCHEMA = getDlqSchema();
+
+  private PubsubResourceManager pubsubResourceManager;
+  private BigQueryResourceManager bigQueryResourceManager;
+
+  @Before
+  public void setUp() throws IOException {
+    pubsubResourceManager =
+        PubsubResourceManager.builder(testName, PROJECT, credentialsProvider).build();
+    bigQueryResourceManager =
+        BigQueryResourceManager.builder(testName, PROJECT, credentials).build();
+
+    gcsClient.createArtifact(
+        "udf.js",
+        "function uppercaseName(value) {\n"
+            + "  const data = JSON.parse(value);\n"
+            + "  data.name = data.name.toUpperCase();\n"
+            + "  return JSON.stringify(data);\n"
+            + "}");
+    gcsClient.createArtifact(
+        "cdc_schema.json",
+        "["
+            + "  {"
+            + "    \"type\": \"INTEGER\","
+            + "    \"name\": \"id\","
+            + "    \"mode\": \"NULLABLE\""
+            + "  }, "
+            + "  {"
+            + "    \"type\": \"STRING\","
+            + "    \"name\":\"job\","
+            + "    \"mode\":\"NULLABLE\""
+            + "  }, "
+            + "  {"
+            + "    \"type\": \"String\","
+            + "    \"name\": \"name\","
+            + "    \"mode\": \"NULLABLE\""
+            + "  }"
+            + "]");
+  }
+
+  @After
+  public void cleanUp() {
+    ResourceManagerUtils.cleanResources(pubsubResourceManager, bigQueryResourceManager);
+  }
+
+  @Test
+  public void testSubscriptionToBigQuery() throws IOException {
+    testSubscriptionToBigQueryBase(false);
+  }
+
+  @Test
+  public void testSubscriptionToBigQueryWithUDf() throws IOException {
+    testSubscriptionToBigQueryBase(true);
+  }
+
+  public void testSubscriptionToBigQueryBase(boolean useUdf) throws IOException {
```

Comment:
the suffix "test" is usually for junit test. Just name "subscriptionToBigQueryBase" ?